--- a/src/Illuminate/Console/ContainerCommandLoader.php
+++ b/src/Illuminate/Console/ContainerCommandLoader.php
@@ -47,8 +47,14 @@ class ContainerCommandLoader implements CommandLoaderInterface
     public function get(string $name): Command
     {
         if (! $this->has($name)) {
             throw new CommandNotFoundException(sprintf('Command "%s" does not exist.', $name));
         }
 
-        return $this->container->get($this->commandMap[$name]);
+        $command = $this->container->get($this->commandMap[$name]);
+
+        if ($command instanceof \Illuminate\Console\Command && method_exists($this->container, 'make')) {
+            $command->setLaravel($this->container);
+        }
+
+        return $command;
     }
 
